### PR TITLE
Nsm switch 

### PR DESCRIPTION
**Pull request by Houston4444**
NSM switch implementation which allows NSM to switch directly from a session to another one without closing and re-launching Ardour.

For better NSM experience, I also change behaviour of  folder argument, it now prefer the last snapshot to the default snapshot (I mean, this is a bug which appends too whit launching ardour6 mysessionfolder).

The main thing to do was to prevent "save or not" dialogs  and any saves during switch (ardour becomes dirty as a JACK port appears or disapears, so we prefer not save after switch order).

### CODE REVIEW

File: gtk2_ardour/ardour_ui.cc [hunk 1]
```diff
@@ -3553,11 +3577,11 @@ ARDOUR_UI::get_session_parameters (bool quit_on_cancel, bool should_be_new, stri
 		} else if (likely_new && template_name.empty()) {
 
 			ret = build_session_from_dialog (session_dialog, session_path, session_name);
-
+
```
Comment (x42) [inline gtk2_ardour/ardour_ui.cc]: Trailing whitespace

File: gtk2_ardour/ardour_ui.cc [hunk 2]
```diff
@@ -3553,11 +3577,11 @@ ARDOUR_UI::get_session_parameters (bool quit_on_cancel, bool should_be_new, stri
 		} else if (likely_new && template_name.empty()) {
 
 			ret = build_session_from_dialog (session_dialog, session_path, session_name);
-
+			
 		} else {
 
 			ret = load_session (session_path, session_name, template_name);
-
+
```
Comment (x42) [inline gtk2_ardour/ardour_ui.cc]: trailing whitespace

File: gtk2_ardour/ardour_ui.cc [hunk 3]
```diff
@@ -3575,7 +3599,13 @@ ARDOUR_UI::get_session_parameters (bool quit_on_cancel, bool should_be_new, stri
 			ARDOUR_COMMAND_LINE::session_name = "";
 		}
 	}
+	
+	if (_nsm_switches && ret != 0 ) {
+			/*If there is an error at nsm open or switch, don't loop an error dialog, just quit. */
+			exit (1);
+	}
 
+	_nsm_switches = false;
```
Comment (x42) [inline gtk2_ardour/ardour_ui.cc]: Is this the right place to reset `_nsm_switches` ?
There are various ways to load or construct a session and not all are via `get_session_parameters()`. I don't know if that applies to NSM though. But perhaps `ARDOUR_UI::set_session()` is a better place to reset it.

Comment (Houston4444) [reply gtk2_ardour/ardour_ui.cc]: Hummm, I don't find where set_session() is called in get_session_parameters().
One thing is sure,  NSM always call get_session_parameters() to open or create a session (see nsm.cc line55). And if ardour isn't launch with NSM, _nsm_switches is never set.

Ardour is set as 'ready' for NSM, once get_session_parameters() is terminate (if there is no errror).

Comment (x42) [reply gtk2_ardour/ardour_ui.cc]: set_session() is called whenever the session changes. AFICT it's the only central method that's always called in that case.

Comment (Houston4444) [reply gtk2_ardour/ardour_ui.cc]: Ok cool. Just a little question about acronym... What means AFICT ???

Comment (x42) [reply gtk2_ardour/ardour_ui.cc]: oops. Sorry, I meant AFAICT -- as far as I can tell.

**Comment by Houston4444** [comment]
Replaced with #651

**Pull request closed without merging**